base/linalg/triangular.jl

```diff
@@ -1900,14 +1907,10 @@ end
 function sqrtm{T}(A::UnitUpperTriangular{T})
```

Comment:
How did you unwrap the type?  I hope you didn't do `A = A.data`, which hurts type stability.

I can't reproduce your benchmark results.  Maybe the benchmarks aren't reliable for such a short operation, and one needs to put a bunch of indexing operations in a loop?  (Probably you also want to time them with `@inbounds`).   The `$` splices the value of the variable directly into the expression, which gets rid of the penalty for benchmarking with a global variable (it means the compiler does not have to do runtime type inference to find the type of `A`).